src/prefect/engine.py

```diff
@@ -795,15 +801,14 @@ async def begin_task_map(
                 wait_for=wait_for,
                 return_type=return_type,
                 task_runner=task_runner,
+                extra_task_inputs=task_inputs,
```

Comment:
I'm not _excited_ about `extra_task_inputs`, but I also can't think of a better approach at the moment. It may point at a possible need to refactor the way we validate the parameters to a map call or the way we gather at task inputs at some point.

Reply:
Yeah me neither. I think ideally we have a way to track lengths on futures and create indexed children of futures. When the future has no known length, we resolve it to determine the length for mapping then access its indices (which creates child futures) that we pass downstream. That way, our existing tracking will work.

That said, users want to create dependencies between tasks in their graph without necessarily _waiting for_ them to finish. `extra_task_inputs` may be useful for user-defined relationships that do not have extra implications.

Reply:
cc @WillRaphaelson for reference